runtime/tests/shared/SharedCacheAPITest.cpp

```diff
@@ -237,16 +237,16 @@ validateSharedCacheCallback(J9JavaVM *vm, J9SharedCacheInfo *cacheInfo, void *us
 		/* J9PORT_SHR_CACHE_TYPE_SNAPSHOT == cacheInfo->cacheType */
 		if ((strcmp(cacheInfo->name, cacheInfoList[NUM_CACHE].name) == 0)
 			&& (cacheInfo->cacheType == cacheInfoList[NUM_CACHE].cacheType)
-			&& (J9SH_OSCACHE_UNKNOWN == (IDATA)cacheInfo->isCorrupt)
+			&& (J9SH_OSCACHE_UNKNOWN == cacheInfo->isCorrupt)
 			&& (1 == cacheInfo->isCompatible)
 			&& (addrMode == cacheInfo->addrMode)
 			&& (getShcModlevelForJCL(J2SE_VERSION(vm)) == cacheInfo->modLevel)
-			&& (J9SH_OSCACHE_UNKNOWN == cacheInfo->lastDetach)
-			&& (J9SH_OSCACHE_UNKNOWN == (IDATA)cacheInfo->os_shmid)
-			&& (J9SH_OSCACHE_UNKNOWN == (IDATA)cacheInfo->os_semid)
-			&& (J9SH_OSCACHE_UNKNOWN == (IDATA)cacheInfo->cacheSize)
-			&& (J9SH_OSCACHE_UNKNOWN == (IDATA)cacheInfo->freeBytes)
-			&& (J9SH_OSCACHE_UNKNOWN == (IDATA)cacheInfo->softMaxBytes)
+			&& ((I_64)J9SH_OSCACHE_UNKNOWN == cacheInfo->lastDetach)
```

Comment:
I think the cast should be on the right:
```
    J9SH_OSCACHE_UNKNOWN == (U_64)cacheInfo->lastDetach
```
The compiler will zero-extend the left-hand-side if necessary.

Reply:
Zero extending won't work, lastDetach will contain all 1's.

Reply:
You're right, but casting on the left is wrong too, it should be:
```
    J9SH_OSCACHE_UNKNOWN == (UDATA)(U_64)cacheInfo->lastDetach
```
so no extension is necessary.

Reply:
Updated.